core/sockets.py

```diff
@@ -86,7 +86,17 @@ def set_default(self, value):
     @property
     def socket_id(self):
         """Id of socket used by data_cache"""
-        return str(hash(self.id_data.name + self.node.name + self.identifier))
+        # if not self.socket_id_m:
+            # try:
+            # self.socket_id_m = str(hash(self) ^ hash(time.monotonic()))
+            # except:
+                # return str(hash(self) ^ hash(time.monotonic()))
+            # self.socket_id_m.set(str(hash(self) ^ hash(time.monotonic())))
+            # self.set("socket_id_m", str(hash(self) ^ hash(time.monotonic())))
+        # return self.socket_id_m
+        # return str(hash(self.id_data.name + self.node.name + self.identifier))
+        return str(hash(self.node.node_id + self.identifier))
```

Comment:
Is it really need to change this method?

Reply:
To support node and tree renaming without losing the track of the data it has to relay on node_id